File: datastructures/stacks/python/algorithms/validate_parentheses.py
def is_valid(s):
    if len(s) < 2:
        return False

    dict = {
        "{": "}",
        "[": "]",
        "(": ")"
    }

    temp = []

    for element in s:
        if len(temp) > 0 and dict.get(temp[len(temp) - 1], None) == element:
            temp.pop()
        else:
            temp.append(element)

    return len(temp) == 0


def main():
    s = "()"

    print(is_valid(s))


if __name__ == "__main__":
    main()

File: datastructures/stacks/java/algorithms/ValidateParentheses.java
import java.util.HashMap;
import java.util.Map;
import java.util.Stack;

public class ValidateParentheses {

    public static boolean isValid(String s) {
        if (s.length() < 2) {
            return false;
        }

        Stack<String> temp = new Stack<>();

        Map<String, String> paranthesesDict = new HashMap<>();
        paranthesesDict.put("{", "}");
        paranthesesDict.put("[", "]");
        paranthesesDict.put("(", ")");

        for (int i = 0; i < s.length(); i++) {
            if (!temp.empty() && paranthesesDict.getOrDefault(temp.peek(), "*").equals(String.valueOf(s.charAt(i)))) {
                temp.pop();
            } else {
                temp.push(String.valueOf(s.charAt(i)));
            }
        }

        return temp.size() == 0;

    }

    public static void main(String[] args) {
        String input = "([)]";

        System.out.println("Is input valid? " + isValid(input));
    }

}
